app/models/move.rb

```diff
@@ -126,17 +126,22 @@ def rebooked
   end
 
   def rebook
-    rebooked || self.class.create(
+    return rebooked if rebooked.present?
+
+    new_booking = self.class.new(
       original_move_id: id,
       from_location_id: from_location_id,
       to_location_id: to_location_id,
       allocation_id: allocation_id,
       profile_id: profile_id,
       status: MOVE_STATUS_PROPOSED,
       date: date && date + 7.days,
-      date_from: date_from && date_from + 7.days,
+      date_from: (date_from || date) + 7.days,
       date_to: date_to && date_to + 7.days,
     )
+    new_booking.determine_supplier
```

Comment:
```suggestion
    new_move.determine_supplier
```

Reply:
I think the `determine_supplier` method is about to disappear, but check with Andrew